### PR TITLE
feat(GOL-1): add unit tests for Material 3 buttons

### PR DESCRIPTION
**Pull request by GUIDOURI**
###   What was done?

- Created unit tests for all button variants (Filled, Elevated, Tonal, Outlined, Text) using React Testing Library.
- Covered edge cases and state changes (e.g., disabled, loading, with icons).
- Integrated animations and design tokens.

###   Why?

To ensure UI components follow expected behavior and appearance under different props and states, increasing maintainability and trust.

###  How to test?

1. Run:
   ```bash
   pnpm test src/components/button
2. All button variants should be tested successfully.

3. Storybook renders should also match the tested components.

### CODE REVIEW

**Comment by ougarte** [comment]
@GUIDOURI I recommend renaming the branch to `features/GOL-1`

File: apps/api/src/app/app.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import type { TestingModule } from '@nestjs/testing';
+import { Test } from '@nestjs/testing';
+import { AppController } from './app.controller';
+import { AppService } from './app.service';
+
+describe('AppController', () => {
```
Comment (ougarte) [inline apps/api/src/app/app.controller.spec.ts:6]: I recommend renaming from `AppController` to `app/controller`

File: apps/api/src/app/app.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	let app: TestingModule;
+
+	beforeAll(async () => {
+		app = await Test.createTestingModule({
+			controllers: [AppController],
+			providers: [AppService],
+		}).compile();
+	});
+
+	describe('getData', () => {
```
Comment (ougarte) [inline apps/api/src/app/app.controller.spec.ts:16]: I recommend removing this `describe`

File: apps/api/src/app/app.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+		it('should return "Hello API"', () => {
```
Comment (ougarte) [inline apps/api/src/app/app.controller.spec.ts:17]: I recommend renaming to `should get a message`, as follow

```ts
test('should get a message', async () => {
    // arrange
    const controller = app.get<AppController>(AppController);

    // act
    const response = await controller.getData();

    // assert
    expect(response).toEqual({ message: 'Hello API' })
});
```

File: apps/ui/src/components/button/types.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import type { ReactNode, ReactElement, HTMLAttributes, ButtonHTMLAttributes } from 'react';
+
+export type IconProperties = HTMLAttributes<{
```
Comment (ougarte) [inline apps/ui/src/components/button/types.ts:3]: I recommend removing `IconProperties`

File: apps/ui/src/components/button/types.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	/**
+	 *
+	 */
+	children?: ReactNode | null;
+}>;
+
+export type TextProperties = HTMLAttributes<{
```
Comment (ougarte) [inline apps/ui/src/components/button/types.ts:10]: I recommend removing `TextProperties`

File: apps/ui/src/components/button/types.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	/**
+	 *
+	 */
+	children?: ReactNode | null;
+}>
+
+/**
+ *
+ */
+export type Properties = {
```
Comment (ougarte) [inline apps/ui/src/components/button/types.ts:20]: I recommend updating as follow
```ts
export type Properties = ButtonHTMLAttributes<HTMLButtonElement> & {
    as?: ReactElement;
    className?: string;
    children?: ReactNode;
}
```

File: apps/ui/src/components/button/styles/button.module.css [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+.text:hover {
+  background-color: var(--md-text-hover);
+}
+
+/* Animación */
+.fadeIn {
+  animation: fadeIn 0.4s ease-in-out both;
+}
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css:87]: I recommend moving `.fabeIn` to `animation.css` file

File: apps/ui/src/components/button/styles/button.module.css [hunk 8]
```diff
@@ -0,0 +1,87 @@
+@import './variables.css';
+@import './animations.css';
+
+/* Estilo base del botón */
+.buttonBase {
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css:5]: I recommend using `.button`

File: apps/ui/src/components/button/styles/button.module.css [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+  border: none;
+  cursor: pointer;
+  transition: background-color 0.3s ease, box-shadow 0.3s ease, transform 0.2s ease;
+  position: relative;
+  overflow: hidden;
+}
+
+.fullWidth {
+  width: 100%;
+}
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css:23]: I recommend removing `.fullWidth` from `button.css` file

File: apps/ui/src/components/button/styles/button.module.css [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+.icon {
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css:25]: I recommend adding a prefix as follow `.button .icon`

File: apps/ui/src/components/button/styles/button.module.css [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+/* Variantes de Material 3 */
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css:31]: I recommend moving all variants into `.button` class as follow 

```css
.button {
    &.filled { ... }
    /* ....more styles */
}
```

File: apps/ui/src/components/button/index.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
+export { default as ButtonComponent } from './button';
+export { default as ButtonTextComponent } from './button.text';
+export { default as ButtonIconComponent } from './button.icon';
+export { default as BaseButton } from './button-base';
+export { default as ButtonFilledComponent } from './button.filled';
+export { default as ButtonElevatedComponent } from './button.elevated';
+export { default as ButtonTonalComponent } from './button.tonal';
+export { default as ButtonOutlinedComponent } from './button.outlined';
+export { default as ButtonFABComponent } from './button.fab';
+export { default as ButtonSegmentedComponent } from './button.segmented';
```
Comment (ougarte) [inline apps/ui/src/components/button/index.ts:10]: I recommend exporting a unique component `export { default as Button } from './button';`

File: apps/ui/src/components/button/index.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+
+// INFO: Exports typings
+export type {
+	Properties as ButtonProperties,
+	IconProperties as ButtonIconProperties,
+	TextProperties as ButtonTextProperties,
+} from './types';
```
Comment (ougarte) [inline apps/ui/src/components/button/index.ts:18]: I recommend exporting a unique type `export type { Properties as ButtonProperties } from './button';`

File: apps/ui/src/components/button/button.tonal.tsx [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import BaseButton, { ButtonBaseProps } from './button-base';
+
+const TonalButton: React.FC<ButtonBaseProps> = (props) => {
+  return <BaseButton variant="tonal" {...props} />;
+};
+
+export default TonalButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tonal.tsx:8]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button.text.tsx [hunk 15]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import BaseButton, { ButtonBaseProps } from './button-base';
+
+const TextButton: React.FC<ButtonBaseProps> = (props) => {
+  return <BaseButton variant="text" {...props} />;
+};
+
+export default TextButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.text.tsx:8]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button.segmented.tsx [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <BaseButton
+      variant="outlined"
+      className={`${selected ? 'selected' : ''} ${className || ''}`}
+      {...props}
+    />
+  );
+};
+
+export default SegmentedButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.segmented.tsx:22]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button.outlined.tsx [hunk 17]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import BaseButton, { ButtonBaseProps } from './button-base';
+
+const OutlinedButton: React.FC<ButtonBaseProps> = (props) => {
+  return <BaseButton variant="outlined" {...props} />;
+};
+
+export default OutlinedButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.outlined.tsx:8]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button.icon.tsx [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+
+const IconButton: React.FC<ButtonBaseProps> = ({ children, ...props }) => {
+  return (
+    <BaseButton variant="text" icon={children} {...props}>
+      {children}
+    </BaseButton>
+  );
+};
+
+export default IconButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.icon.tsx:12]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button.filled.tsx [hunk 19]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import BaseButton, { ButtonBaseProps } from './button-base';
+
+const FilledButton: React.FC<ButtonBaseProps> = (props) => {
+  return <BaseButton variant="filled" {...props} />;
+};
+
+export default FilledButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.filled.tsx:8]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button.fab.tsx [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+      icon={icon}
+      aria-label="Floating Action Button"
+      {...props}
+    >
+      {children}
+    </BaseButton>
+  );
+};
+
+export default FABButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.fab.tsx:18]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button.elevated.tsx [hunk 21]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import BaseButton, { ButtonBaseProps } from './button-base';
+
+const ElevatedButton: React.FC<ButtonBaseProps> = (props) => {
+  return <BaseButton variant="elevated" {...props} />;
+};
+
+export default ElevatedButton;
```
Comment (ougarte) [inline apps/ui/src/components/button/button.elevated.tsx:8]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/button-base.tsx [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+
+  return (
+    <button className={classes} {...props}>
+      {icon && <span className={styles.icon}>{icon}</span>}
+      <span>{children}</span>
+    </button>
+  );
+};
+
+export default ButtonBase;
```
Comment (ougarte) [inline apps/ui/src/components/button/button-base.tsx:41]: I recommend moving this implementation into `button.tsx`

File: apps/ui/src/components/button/__tests__/button.elevated.test.tsx [hunk 23]
```diff
@@ -0,0 +1,16 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import ElevatedButton from '../button.elevated'; // asegúrate de que el componente se exporte como default
+
+describe('ElevatedButton', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.elevated.test.tsx:5]: I recommend moving all tests to `button.test.tsx`

File: apps/ui/src/components/button/__tests__/button.filled.test.tsx [hunk 24]
```diff
@@ -0,0 +1,16 @@
+import { render, screen } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import FilledButton from '../button.filled'
+
+describe('FilledButton', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.filled.test.tsx:5]: I recommend moving all tests to `button.test.tsx`

File: apps/ui/src/components/button/__tests__/button.icon.test.tsx [hunk 25]
```diff
@@ -0,0 +1,17 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import IconButton from '../button.icon';
+
+describe('IconButton', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.icon.test.tsx:5]: I recommend moving all tests to `button.test.tsx`

File: apps/ui/src/components/button/__tests__/button.outlined.test.tsx [hunk 26]
```diff
@@ -0,0 +1,16 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import OutlinedButton from '../button.outlined';
+
+describe('OutlinedButton', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.outlined.test.tsx:5]: I recommend moving all tests to `button.test.tsx`

File: apps/ui/src/components/button/__tests__/button.text.test.tsx [hunk 27]
```diff
@@ -0,0 +1,16 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import TextButton from '../button.text';
+
+describe('TextButton', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.text.test.tsx:5]: I recommend moving all tests to `button.test.tsx`

File: apps/ui/src/components/button/__tests__/button.tonal.test.tsx [hunk 28]
```diff
@@ -0,0 +1,16 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import TonalButton from '../button.tonal';
+
+describe('TonalButton', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.tonal.test.tsx:5]: I recommend moving all tests to `button.test.tsx`

File: apps/ui/src/components/button/__tests__/button.test.tsx [hunk 29]
```diff
@@ -0,0 +1,7 @@
+// describe('components/button', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.test.tsx:1]: I recommend updating as follow:


```tsx
describe('components/button', () => {
    describe('layout', () => {
         test('should render by default', () => { });
         test('should render text variant by default', () => { });
         test('should render outline variant by default', () => { });
         // ... more test
        
         test('should render an icon', () => { });
    });

    describe('behavior', () => {
         test('should trigger click event', () => { });
    });
})
```
